File: interview_code/dealing_with_strings/letter_freq.js
// Completed, not refactored

function letterFrequency(string_for_counting) {
    var string_as_array = string_for_counting.split("");
    var array_string_length = string_as_array.length;
    var frequency_hash = {};
    for(var i = 0; i < array_string_length; i++){  //loop through array
        if(frequency_hash.hasOwnProperty(string_as_array[i])) {
            frequency_hash[string_as_array[i]] +=1;
        } else {
            frequency_hash[string_as_array[i]] = 1;
        }
    }
    return frequency_hash;
}

var a = letterFrequency("mississippi river is running after me");
console.log(a);

File: interview_code/dealing_with_strings/letter_freq.rb
# Completed, not refactored

def frequency_counter(input_string)
    @input_chars = input_string.split("")
    @freq_hash = Hash.new(0)
    @input_chars.each do |letter|
        @freq_hash[letter]+=1
    end
    return @freq_hash
end

a = "mississippi is not here"
frequency_counter(a)
